sunkit_instruments/goes_xrs/goes_chianti_tem.py

```diff
@@ -0,0 +1,317 @@
+import numpy as np
+import pandas as pd
+from scipy import interpolate
+
+from astropy import units as u
+from astropy.io import fits
+from astropy.time import Time
+from sunpy import timeseries as ts
+from sunpy.data import manager
+from sunpy.time import parse_time
+from sunpy.util.exceptions import warn_user
+
+__all__ = ["calculate_temperature_em"]
+
+
+def calculate_temperature_em(goes_ts, abundance="coronal"):
+    """
+    This function calculates the isothermal temperature and
+    corresponding volume emission measure of the solar soft X-ray
+    emitting plasma observed by GOES/XRS.
+
+    These are calculated based on methods described in White et al. 2005 [1]_ (see notes)
+    for which the GOES fluxes and channel ratios are used together with look-up tables
+    of CHIANTI atomic models to estimate isothermal temperature and emission measure.
+    Technically speaking, the method interpolates on the channel flux ratio using
+    pre-calcuated tables for the fluxes at a series of temperatures for fixed emission
+    measure.
+
+    The method here is almost an exact replica of what is available in SSWIDL,
+    namely, goes_chianti_tem.pro, and it has been tested against that for consistency.
```

Comment:
Do we have a test for this?

Reply:
no we dont have an explicit one. 

maybe i'll just remove this sentence and say that the code is the exact replica.

Reply:
not much, I'll add it here.

Reply:
now added

Reply:
Thanks